rllib/core/sarl_trainer.py

```diff
@@ -0,0 +1,34 @@
+class SARLTrainer:
+    def train(self, batch):
+        # this is the function is called by the user to train the model
+        for queue in self.queues:
+            queue.put(batch)
+        results = next(self.training_iterator)
+        self.curr_weights = results[0]["module_weights"]
+        return results
+
+    @staticmethod
+    def compute_loss(batch, fwd_out, **kwargs):
+        raise NotImplementedError
+
+    @staticmethod
+    def compute_grads_and_apply_if_needed(batch, fwd_out, loss_out, **kwargs):
```

Comment:
I'm not a fan of functions that __might__ have side effects. Is there a reason we don't have a separate `compute_grads` and `apply_grads` function?